### PR TITLE
[search] Expose num typos to ranking info.

### PR DESCRIPTION
**Pull request by ygorshenin**
@mpimenov PTAL

### CODE REVIEW

File: search/search_integration_tests/helpers.cpp [hunk 1]
```diff
@@ -79,11 +79,18 @@ bool SearchTest::ResultsMatch(SearchParams const & params, TRules const & rules)
   return ResultsMatch(request.Results(), rules);
 }
 
-unique_ptr<tests_support::TestSearchRequest> SearchTest::MakeRequest(string const & query)
+bool SearchTest::ResultMatches(search::Result const & result,
```
Comment (mpimenov) [inline search/search_integration_tests/helpers.cpp]: Why do you need this?

File: search/geocoder.cpp [hunk 2]
```diff
@@ -353,7 +353,7 @@ void Geocoder::SetParams(Params const & params)
         // Here and below, we use LevenshteinDFAs for fuzzy
         // matching. But due to performance reasons, we assume that the
         // first letter is always correct.
```
Comment (mpimenov) [inline search/geocoder.cpp]: Move the comment to BuildLevenshteinDFA, please.

File: search/ranking_utils.hpp [hunk 3]
```diff
@@ -18,11 +21,64 @@ namespace search
...
+      return lhs;
+    return ErrorsMade(fn(lhs.m_errorsMade, rhs.m_errorsMade));
+  }
+
+  static ErrorsMade Min(ErrorsMade const & lhs, ErrorsMade const & rhs)
+  {
+    return Combine(lhs, rhs, [](size_t u, size_t v) { return std::min(u, v); });
+  }
+
+  static ErrorsMade Max(ErrorsMade const & lhs, ErrorsMade const & rhs)
```
Comment (mpimenov) [inline search/ranking_utils.hpp]: Why do you need this?

File: search/ranker.cpp [hunk 4]
```diff
@@ -12,33 +12,39 @@
 
 #include "std/algorithm.hpp"
 #include "std/iterator.hpp"
+#include "std/limits.hpp"
```
Comment (mpimenov) [inline search/ranker.cpp]: Why do you need this?

File: search/ranking_utils.hpp [hunk 5]
```diff
@@ -92,4 +148,30 @@ NameScore GetNameScore(std::vector<strings::UniString> const & tokens, Slice con
 }
 
 string DebugPrint(NameScore score);
+
+// Returns maximum number of errors that were made during matching
```
Comment (mpimenov) [inline search/ranking_utils.hpp]: What do you mean by "maximum" here?

Comment (ygorshenin) [reply search/ranking_utils.hpp]: "maximum" is by mistake here. The correct word is "total".

File: search/ranker.cpp (hunk 4 above)
Comment (ygorshenin) [reply search/ranker.cpp]: Fixed.

File: search/ranking_utils.hpp (hunk 3 above)
Comment (ygorshenin) [reply search/ranking_utils.hpp]: Old code, not needed now.

File: search/search_integration_tests/helpers.cpp (hunk 1 above)
Comment (ygorshenin) [reply search/search_integration_tests/helpers.cpp]: Old code, not needed now.

File: search/geocoder.cpp (hunk 2 above)
Comment (ygorshenin) [reply search/geocoder.cpp]: Done

**Comment by ygorshenin** [comment]
@mpimenov PTAL

**Pull request merged**